repos/system_upgrade/el7toel8/actors/localreposinhibit/actor.py

```diff
@@ -0,0 +1,83 @@
+from typing import List
+
+from leapp import reporting
+from leapp.actors import Actor
+from leapp.models import (
+    CustomTargetRepository,
+    TargetRepositories,
+    UsedTargetRepositories,
+    UsedTargetRepository,
+)
+from leapp.reporting import Report
+from leapp.tags import IPUWorkflowTag, TargetTransactionChecksPhaseTag
+
+
+class LocalReposInhibit(Actor):
+    """Inhibits the upgrade if local repositories were found.
+
+    Currently we're not supporting local repositories during the upgrade.
+    There is a workaround available, which described inside the remediation of
+    the report.
+
+    We're looking for a baseurl only inside TargetRepositories.custom_repos
+    because local repos might exists as only custom repository, due to the
+    fact that they might be specified only inside custom repo file of the leapp
+    or via --enablerepo flag. Both options puts the repo into
+    CustomTargetRepository.
+    """
+
+    name = "local_repos_inhibit"
+    consumes = (UsedTargetRepositories, TargetRepositories)
+    produces = (Report,)
+    tags = (IPUWorkflowTag, TargetTransactionChecksPhaseTag)
+
+    def process(self):
+        used_target_repos = next(
+            self.consume(UsedTargetRepositories)
+        ).repos  # type: List[UsedTargetRepository]
+        custom_target_repos = next(
+            self.consume(TargetRepositories)
+        ).custom_repos  # type: List[CustomTargetRepository]
+        for custom_repo in filter(  # pylint: disable-msg=deprecated-lambda
+            lambda custom_target_repo: custom_target_repo.repoid
+            in (used_tg_repo.repoid for used_tg_repo in used_target_repos)
+            and custom_target_repo.baseurl,
+            custom_target_repos,
+        ):
```

Comment:
This is super compact, but super unreadable. I suggest using a small and simple function instead:
```suggestion
        for custom_repo in _get_custom_repos():
```
Remember, clean code has functions with a few lines. And each function has a name that describes what it does. Then, you can read the function calls as easily as a story in a book.
We can debate what number of lines is optimal/maximum for a function, whether 10, 20 or 30. But this one has 50 lines.

Reply:
Thanks for advising on this